src/components/app-controllers/FloodAppController/FloodAppController.js

```diff
@@ -0,0 +1,231 @@
+                      value={this.props.variable}
+                      onChange={this.props.onChangeVariable}
+                      replaceInvalidValue={this.replaceInvalidVariable}
+                    />
+                  </Col>
+                </Row>
+              </Panel.Body>
+            </Panel>
+          </FullWidthCol>
+        </Row>
+
+        <Row>
+          <FullWidthCol>
+            <FlowArrow pullUp />
+          </FullWidthCol>
+        </Row>
+
+        <Row>
+          <FullWidthCol>
+            <FilteredDatasetsSummary
+              model_id={model_id}
+              experiment={experiment}
+              variable_id={variable_id}
+              meta = {filteredMeta}
+            />
+          </FullWidthCol>
+        </Row>
+
+        <Row>
+          <HalfWidthCol>
+            <FlowArrow pullUp />
+          </HalfWidthCol>
+          <HalfWidthCol>
+            <FlowArrow pullUp />
+          </HalfWidthCol>
+        </Row>
+
+        <Row>
+          <HalfWidthCol>
+            <SingleMapController
+              model_id={model_id}
+              experiment={experiment}
+              variable_id={variable_id}
+              meta = {filteredMeta}
+              area={this.props.area}
+              onSetArea={this.props.onChangeArea}
+              pointSelect={true}
+            />
+          </HalfWidthCol>
+          <HalfWidthCol>
+            <FloodDataController
+              ensemble_name={this.props.ensemble_name}
+              model_id={model_id}
+              variable_id={variable_id}
+              experiment={experiment}
+              area={g.geojson(this.props.area).toWKT()}
+              meta = {filteredMeta}
+              contextMeta={modelContextMetadata} //to generate Model Context graph
+            />
+          </HalfWidthCol>
+        </Row>
+      </Grid>
+    );
+  }
+}
+
+
+// Inject asynchronously fetched metadata into controlled component.
+const WmdFloodAppControllerDisplay = withAsyncMetadata(FloodAppControllerDisplay);
+
+
+export default class FloodAppController extends React.Component {
+  // This manages the state of selectors and renders the display component.
+
+  state = {
+    model: undefined,
+    scenario: undefined,
+    variable: undefined,
+    area: undefined,  // geojson object
+  };
+
+  // TODO: https://github.com/pacificclimate/climate-explorer-frontend/issues/122
+  handleChangeArea = setNamedState(this, 'area');
+  handleChangeModel = setNamedState(this, 'model');
+  handleChangeScenario = setNamedState(this, 'scenario');
+  handleChangeVariable = setNamedState(this, 'variable');
+
+  render() {
+    return (
+      <WmdFloodAppControllerDisplay
+        ensemble_name={ensemble_name(this.props)}
+        {...this.state}
+        onChangeArea={this.handleChangeArea}
+        onChangeModel={this.handleChangeModel}
+        onChangeScenario={this.handleChangeScenario}
+        onChangeVariable={this.handleChangeVariable}
+      />
+    );
+  }
+}
```

Comment:
Plausible, boilerplate. Anything that might need a closer look?

Reply:
Not really, it's all copy and paste from SingleAppController.